server/channels/app/helper_test.go

```diff
@@ -724,3 +728,24 @@ func NewTestId() string {
 func (th *TestHelper) NewPluginAPI(manifest *model.Manifest) plugin.API {
 	return th.App.NewPluginAPI(th.Context, manifest)
 }
+
+func decodeJSON[T any](o any, result *T) *T {
+	var r io.Reader
+	switch v := o.(type) {
+	case string:
+		r = strings.NewReader(v)
+	case []byte:
+		r = bytes.NewReader(v)
+	case io.Reader:
+		r = v
+	default:
+		panic(fmt.Sprintf("Unable to decode JSON from %v", v))
```

Comment:
nit: would it be useful to log the type here, with the value as secondary?
```suggestion
		panic(fmt.Sprintf("Unable to decode JSON from %T (%v)", v, v))
```